arch/risc-v/src/esp32c3/esp32c3_wlan.c

```diff
@@ -691,27 +486,9 @@ static void wlan_rxpoll(void *arg)
       pkt_input(&priv->dev);
 #endif
 
-      /* Check if the packet is a valid size for the network
-       * buffer configuration (this should not happen)
-       */
-
-      if (dev->d_len > WLAN_BUF_SIZE)
-        {
-          nwarn("WARNING: DROPPED Too big: %d\n", dev->d_len);
-
-          /* Free dropped packet buffer */
-
-          if (dev->d_buf)
-            {
-              wlan_free_buffer(priv, dev->d_buf);
-              dev->d_buf = NULL;
-              dev->d_len = 0;
-            }
-
-          continue;
-        }
-
-      eth_hdr = (struct eth_hdr_s *)dev->d_buf;
+      eth_hdr = (struct eth_hdr_s *)
+        &dev->d_iob->io_data[CONFIG_NET_LL_GUARDSIZE -
+                             NET_LL_HDRLEN(dev)];
```

Comment:
```suggestion
      eth_hdr = ETHBUF;
```

Reply:
ETHBUF was moved to internal header, so ETHBUF can no longer be used from arch driver  :)
https://github.com/apache/nuttx/blob/62004a28a68f47f0c6f31c9e050555997bfbca06/net/devif/devif.h#L240

Reply:
ok.